### PR TITLE
marshal endpoint

### PR DESCRIPTION
**Pull request by alexander-chervony**

### CODE REVIEW

File: gateway.go [hunk 1]
```diff
@@ -109,6 +113,62 @@ func (s *gatewayResource) gatewayHandler(w http.ResponseWriter, r *http.Request)
 	w.Write(packedResponse)
 }
 
+func (s *gatewayResource) marshalHandler(w http.ResponseWriter, r *http.Request) {
```
Comment (chris-wood) [inline gateway.go:116]: With the new types of handlers, I would instead turn this into an `AppContentHandler` that returns the marshalled proto message. Maybe something like this?

```
type MarshalEchoAppHandler struct {
}

// Handle returns the input request as the response.
func (h MarshalEcho) Handle(binaryRequest []byte, metrics Metrics) ([]byte, error) {
        _, err := requestToProtoHTTP(parsedReq)
        if err != nil {
                return nil, err
        }
        response := &Response{}
        respEnc, err := proto.Marshal(response)
        if err != nil {
                return nil, err
        }
	metrics.Fire(metricsResultSuccess)
	return respEnc, nil
}
```

**Review by chris-wood — CHANGES_REQUESTED** [review]
If the intent of this new handler is to do everything _except_ try to fetch the target resource, I think there's a better way to implement it (as suggested in the comment).

File: gateway.go (hunk 1 above)
Comment (alexander-chervony) [reply gateway.go:122]: thanks for the suggested refactoring! I will get back to it hopefully next week. Fir now I'm running load tests from the branch

Comment (alexander-chervony) [reply gateway.go:127]: Hey Chris! I finally got back to this after error handling refactoring. I've introduced MarshalEncapsulationHandler and MarshalAppHandler as well as a lot of comments for this voodoo-magic code. Hopefully it made things clearer for the poor reader.

**Comment by izhamoidsin** [comment]
The team agreed that the current code is not ready for merge. We're ok with declining the PR. Once we need this marshaling, we'll be able to reissue PR with additional changes.

**Comment by chris-wood** [comment]
Closing per @izhamoidsin's comment.

**Pull request closed without merging**